.github/workflows/clippy.yml

```diff
@@ -0,0 +1,23 @@
+name: Clippy
+
+on:
+  push:
+    branches:
+      - master
+      - staging
+      - trying
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain from rust-toolchain.toml
+        run: rustup show
+      - name: Clippy
+        run: cargo clippy --all-targets -- -D warnings
```

Comment:
I don't really know what the best way is, but my personal preference is, not to create a new file for every job. Therefore I'd rather put it in here: https://github.com/hermitcore/uhyve/blob/master/.github/workflows/test.yml

Reply:
I did it this way in all hermitcore projects. If we change it here, we shall change it in the other ones as well.

Also I could see, why we would be putting it in a `ci.yml` workflow, but it does not fit in `test.yml`, as we don't run tests but check for lints in this job.

Reply:
Alternatively, you could combine it with the format workflow. It is closely related to that, isn't it?

Reply:
I wouldn't say so. Formatting is purely syntactical while the lints are purely semantical.

Reply:
Hmm, I don't fully agree, but as one way is as valuable as the other, let's merge it as is